### PR TITLE
Use the pypi api for distutils3 builds

### PR DESCRIPTION
**Pull request by bryteise**
Get metadata about the package from pypi for distutils3 build patterns
then use that metadata to get summary, description and provides
information (license and requires maybe leveraged in the future).

The description isn't always very useful (often too long) so only use
it if there isn't one already.

### CODE REVIEW

File: autospec/buildreq.py [hunk 1]
```diff
@@ -901,6 +906,27 @@ def scan_for_configure(dirn):
     else:
         config.autoreconf = False
 
+    if buildpattern.default_pattern == "distutils3":
+        # Try and grab the pypi details for the package
+        pypi_api_url = f"https://pypi.python.org/pypi/{tname}/{tversion}/json"
+        pypi_json = download.do_curl(pypi_api_url)
+        if pypi_json:
+            package_pypi = json.loads(pypi_json.getvalue().decode("utf-8"))
```
Comment (bwarden) [inline autospec/buildreq.py]: Should we catch `JSONDecodeError` in case pypi gave us garbage?

File: autospec/buildreq.py [hunk 2]
```diff
@@ -901,6 +906,27 @@ def scan_for_configure(dirn):
...
+            if package_pypi.get("info"):
+                info = package_pypi["info"]
+                if info.get("name"):
+                    pypi_provides = info["name"]
+                if info.get("requires_dist"):
+                    # Not yet adding requires
+                    pass
+                if info.get("license"):
+                    # The license field is freeform, might be worth adding though
```
Comment (bwarden) [inline autospec/buildreq.py]: Maybe print it to the console as a suggestion.

Comment (bryteise) [reply autospec/buildreq.py]: That seems reasonable enough to just do, thanks!

File: autospec/buildreq.py (hunk 1 above)
Comment (bryteise) [reply autospec/buildreq.py]: Seems like a safe thing to do.

**Pull request merged**